hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/MessageContent.java

```diff
@@ -0,0 +1,59 @@
+package pl.allegro.tech.hermes.common.message;
+
+import org.apache.avro.Schema;
+import pl.allegro.tech.hermes.api.ContentType;
+import pl.allegro.tech.hermes.schema.CompiledSchema;
+
+import java.util.Optional;
+
+public class MessageContent {
+    private final ContentType contentType;
+    private final Optional<CompiledSchema<Schema>> schema;
+    private final byte[] data;
+
+    public MessageContent(byte[] content,
```

Comment:
I guess private if we opt for the builder pattern. At present the builder is useless, especially as you still don't enforce schema is not null. Make it default to Optional.empty() - but on the other hand, you could just do the work in the constructor.